.github/actions/analyze-test-runs/src/helpers.js

```diff
@@ -0,0 +1,123 @@
+  // Matches the test class path
+  const TEST_CLASS_PATH_REGEX = /([a-zA-Z][a-zA-Z0-9]*\.)+[a-zA-Z][a-zA-Z0-9]*\.[A-Z][a-zA-Z0-9]*(?:Test|IT)/;
+
+  const match = testLine.match(TEST_CLASS_PATH_REGEX);
+  const result = match ? match[0] : null;
+  console.log(`Extracting test class from: "${testLine}" → ${result || 'NO MATCH'}`);
+  return result;
+}
+
+/**
+ * Generates possible file paths for a given test class
+ * @param {string} testClass - The test class name
+ * @returns {string[]} - Array of possible file paths
+ */
+function generatePossiblePaths(testClass) {
+  const classPath = testClass.replace(/\./g, '/');
+  const paths = [
+    `${classPath}.java`,
+    `src/test/java/${classPath}.java`,
+    `src/main/java/${classPath}.java`
+  ];
+  console.log(`Generated possible paths for ${testClass}:`, paths);
+  return paths;
+}
+
+/**
+ * Finds a matching file for a test class among changed files
+ * @param {string} testClass - The test class name
+ * @param {string[]} changedFiles - Array of changed file paths
+ * @returns {string|null} - The matching file path or null if not found
+ */
+function findMatchingFile(testClass, changedFiles) {
+  const possiblePaths = generatePossiblePaths(testClass);
+  const className = testClass.split('.').pop();
+
+  for (const possiblePath of possiblePaths) {
+    const matchingFile = changedFiles.find(file =>
+      file.endsWith(possiblePath) || path.basename(file) === `${className}.java`
+    );
+    if (matchingFile) {
+      console.log(`Found matching file: ${matchingFile} for test class ${testClass}`);
+      return matchingFile;
+    }
+  }
+  console.log(`No matching file found for test class ${testClass}`);
+  return null;
+}
+
+/**
+ * Creates the inline comment body for a flaky test
+ * @param {string} author - The author to mention
+ * @param {string} flakyTests - The flaky tests string
+ * @returns {string} - The formatted comment body
+ */
+function createInlineCommentBody(author, flakyTests) {
+  return `### Oh! We've got ourselves flaky ones!
+
+Hey @${author}! 👋
+
+This file contains the flaky test:
+\`\`\`
+${flakyTests}
+\`\`\`
+
+Since you recently modified this file, would you mind taking a quick look?
+You might have insights into what could be causing the test instability.
+
+**What to check:**
+- Recent changes that might affect test timing
+- New dependencies or configurations
+- Race conditions or timing-sensitive code
+
+No pressure if you're busy, just thought you'd be the best person to investigate! 🕵️‍♂️`;
+}
+
+/**
+ * Creates the main PR comment body
+ * @param {string} flakyTests - The flaky tests string
+ * @returns {string} - The formatted comment body
+ */
+function createMainCommentBody(flakyTests) {
+  return `### Oh! We've got ourselves flaky ones!
+
+\`\`\`
+${flakyTests}
+\`\`\`
+
+Related to your code? Maybe.
+
+Fixable now? Maybe not.
+
+Please consider investigating if you have time. These tests might return, and
+
+I’ll be back. 🤖`;
```

Comment:
I'd like to have a clearer messaging here: do we want people to take action or not? The "...if you have time" could make most engineers say "I don't" even though their PR might cause the flakiness.

I think it makes sense to ask PR authors to check on whether their PR could cause this flakiness: if yes, then they should address it immediately before merging (blocker); if not, they should leave at least a comment explaining why. WDYT?

Reply:
> This broader message, on the other hand, might not be related to anything made in the task, like in this PR, for example.

Yes I can see that, but also I think flaky tests do not necessary arise from adjustments to the test code. Changing seemingly unrelated files in other modules/setup code/etc can have an impact as well.

> That said, I agree it's worth making the expectations clearer. Maybe we can tweak the wording to suggest that authors should self-assess whether their changes could affect the flaky area, and if so, either fix or explain why not, as mentioned. WDYT?

I think this would be a very good change indeed!